docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,189 @@
+---
+title: "Prompt engineering in .NET"
+description: "Learn basic prompt engineering concepts for .NET."
+author: catbutler
+ms.topic: concept-article #Don't change.
+ms.date: 04/10/2024
+
+#customer intent: As a .NET developer, I want to understand prompt engineering so that I can build more efficient and effective AI apps.
+
+---
+
+# Prompt engineering in .NET
+
+This article covers basic GPT prompt engineering for .NET, including Semantic Kernel and Azure OpenAI.
+
+GPT models from OpenAI are prompt-based: they respond to user input text (a prompt) with the most likely series of words to follow (a completion). The basic completion in earlier GPT models is text generated in response to prompts, producing a more free-form interaction. Newer GPT models produce chat completions that use a specific format, with messages based on roles (system, user, assistant) and chat history to preserve conversations.
+
+Consider this text generation example where prompt is the user input and completion is the model ouput:
+
+Prompt:
+**"The president who served the shortest term was "**
+
+Completion:
+_"Pedro Lascurain."_
+
+Looks right, but what if your app is supposed to help US History students? Pedro Lascurain's 45-minute term is the shortest term for any president, but he served Mexico&mdash;the students are probably looking for _"William Henry Harrsion."_ Clearly, the app could be more helpful to its intended users if you gave it some context.
+
+That's the basic idea of prompt engineering: you add context to the prompt to help the model produce better completions. You can do this by giving the model instructions, examples and cues.
+
+## Instructions
+
+This section explains the use of instructions in prompt engineering.
+
+An instruction is text that tells the model how to respond. An instruction can be a directive (**"You are helping students learn about US history, so talk about the US unless they specifically ask about other countries."**) or an imperative (**"Translate to Tagalog:"**).
+
+Directives are more flexible than imperatives:
+
+- A directive can provide more context than an imperative.
+- You can combine several directives in one instruction.
+- It's usually better to implement a series of steps using a sequence of directives. If you tell the model to output the result of each step, when a problem arises you can easily see which step caused it. Although you can tell the model exactly what steps to follow, you can also just tell the model to break the instruction into steps itself, and tthen to output the result of each step. This approach is called [chain of thought prompting](chain-of-thought-prompting.md).
+
+Instructions are typically more effective when used with examples. However, when you use both in a prompt you should make sure that the instructions are either above or below the examples for best model performance.
```

Comment:
Me too! I'll revise once I understand better. The idea was don't interleave them because you might confuse the model. But there's also other material that says things at the top and bottom of a prompt get more attention from the model than things in the middle, so perhaps the idea here is that instructions require more attention than examples?